### ISSUE TITLE
Friends joining doesn't work on 1.19.40.02

### ISSUE THREAD

**Issue by VDavid003** [minecraft-linux/mcpelauncher-manifest#726]
**Describe the bug**
When on 1.19.40.02, logged in via MS account, friends can't join your world, as they can't see you playing.
You also can't join friend's worlds as they result in an infinite "connecting to server"

**To Reproduce**
1. Start 1.19.40.02
2. Try to join a friend's world
3. Or try starting a world and making an other account join it

**Expected behavior**
Friends can join and I can join friend's worlds

**Desktop (please complete the following information):**
 - OS: Arch Linux (x64)
 - Version: Latest git (ng branch), and every submodule pulled to latest version of their respective branches
 - Installation Type: Built from source

**Additional context**
Haven't tested on LAN or with public servers, those may or may not also be broken.

**Comment by Noggin01** [comment]
Bit off topic, but how did you launch 1.19.40.02? I just bought Minecraft for Windows, PS4, and Android on Saturday. I want my 5 year old on my PS4, my 7 year old on my Windows machine, and myself on my Linux box. It all worked fine yesterday, but my PS4 updated Minecraft to 1.19.40. I was able to update the server I set up and the Windows box, but now I cannot get Minecraft to launch on my Linux box as the launcher says it doesn't support 1.19.40.

**Comment by KoenKDev** [comment]
Yes, I am also interested in how you managed launched 1.19.40.02

**Comment by nebulimity** [comment]
> but how did you launch 1.19.40.02?

I think you have to go into the launcher settings and click "show incompatible versions."

Edit: I meant unverified.

**Comment by VDavid003** [comment]
@Noggin01

> I think you have to go into the launcher settings and click "show incompatible versions."

This may be it, and using the latest git version (with updated submodules for the client) of both the client and launcher helps too.

**Comment by ChristopherHX** [comment]
Before I actually try to follow your reproduction steps just a view questions?
* Is this a new problem caused by updating the Game from 1.19.30 to 1.19.40.02?
* Do you face the same problem on 1.19.30 or lower?

> infinite "connecting to server"

This remembers me that even the win10 edition had trouble to connect to another game if they are not in the same network e.g. Different Internet Access Points.

**Comment by ChristopherHX** [comment]
Toggleing dev options can lead to surprising behavior.
> click "show incompatible versions."

You should use "show unverified versions" instead, because incompatible includes beta and **arm (intel device) / intel (arm device)** versions of the game, which clearly cause more error messages.

**Comment by VDavid003** [comment]
@ChristopherHX 
> Before I actually try to follow your reproduction steps just a view questions?
> 
> * Is this a new problem caused by updating the Game from 1.19.30 to 1.19.40.02?
> * Do you face the same problem on 1.19.30 or lower?

Yes, this seems to be a new problem. On 1.19.31, the last update before this, it worked fine in the same situation (other account on other network (other public IPv4) on Windows 10 connected to my account on Arch Linux).  I don't know about anything changing other than the version on both computers. I did not test extensively though, I did try restarting both clients multiple times and even renaming .local/shared/mcpelauncher to reset some stuff. 

> 
> > infinite "connecting to server"
> 
> This remembers me that even the win10 edition had trouble to connect to another game if they are not in the same network e.g. Different Internet Access Points.

Interesting, I remember having the opposite issue actually between an Android phone and a Windows 10 device. Connection issues on LAN, but perfectly working outside LAN.

**Comment by ChristopherHX** [comment]
I cannot login 100% anymore even on older minecraft versions like 1.17.30, 1.19.30, 1.19.40 1.19.50.21.
You see something like "Opps your ms account got disconnected" in singleplayer worlds. Even after sign out and sign in again. This incomplete login seem to also cause timeouts to connect to featured servers.

_I have no absolutly no idea what happened here, the login flow worked almost 2 years_

**Comment by VDavid003** [comment]
> I cannot login 100% anymore even on older minecraft versions like 1.17.30, 1.19.30, 1.19.40 1.19.50.21. You see something like "Opps your ms account got disconnected" in singleplayer worlds. Even after sign out and sign in again. This incomplete login seem to also cause timeouts to connect to featured servers.

I see that message too if you mean in the ESC menu on the right side below the player list, and I remember seeing it on 1.19.31, but friends connecting still somehow magically worked there. Not sure how this all works in the background.

**Comment by iDarknessSam** [comment]
I'm having this same problem, on the launcher on the version of appimage my friends can't join on my world and the minecraft says that i'm offline even if i am connected on the internet...it's looks like the flatpak version does't occur this problem but i'm having problems of crash on the flatpak version

**Comment by microxxd** [comment]
I have the same problem on the flatpak version and none  of my skins load and i cant join my friends world but am able to join servers

**Comment by brauliobo** [comment]
same here. works with 1.19.30.04 and not with 1.19.51.01.
although with 1.19.30.04 I get incompatible error problem
